airflow/docker_image.go

```diff
@@ -299,7 +299,8 @@ var cmdExec = func(cmd string, stdout, stderr io.Writer, args ...string) error {
 func useBash(authConfig *cliTypes.AuthConfig, image string) error {
 	var err error
 	if authConfig.Username != "" { // Case for cloud image push where we have both registry user & pass, for software login happens during `astro login` itself
-		err = cmdExec(EchoCmd, nil, os.Stderr, fmt.Sprintf("%q", authConfig.Password), "|", DockerCmd, "login", authConfig.ServerAddress, "-u", authConfig.Username, "--password-stdin")
+		cmd := "echo \"" + authConfig.Password + "\"" + " | docker login " + authConfig.ServerAddress + " -u " + authConfig.Username + " --password-stdin"
+		err = cmdExec("bash", os.Stdout, os.Stderr, "-c", cmd)
```

Comment:
will this work on a machine that does not have bash, let's say windows powershell?
what is the root cause for the original command not working with Colima?

Reply:
I don't think that colima works with windows powershell

Reply:
with the old command I was getting `Error: Cannot perform an interactive login from a non TTY device`

Reply:
for some reason running the echo command with bash fixes it: https://stackoverflow.com/questions/52280905/golang-exec-command-bash-command-not-working

Reply:
let's leave a comment in the code on the drawback of the new logic